src/core/net/srp_client.cpp

```diff
@@ -385,7 +385,14 @@ Error Client::Start(const Ip6::SockAddr &aServerSockAddr, Requester aRequester)
                  error = (aServerSockAddr == GetServerAddress()) ? kErrorNone : kErrorBusy);
 
     SuccessOrExit(error = mSocket.Open());
-    SuccessOrExit(error = mSocket.Connect(aServerSockAddr));
+
+    error = mSocket.Connect(aServerSockAddr);
+    if(error != kErrorNone)
+    {
+        LogInfo("Failed to connect to server %s: %s", aServerSockAddr.GetAddress().ToString().AsCString(), ErrorToString(error));
```

Comment:
To fix the pretty check:
```suggestion
    if (error != kErrorNone)
    {
        LogInfo("Failed to connect to server %s: %s", aServerSockAddr.GetAddress().ToString().AsCString(),
                ErrorToString(error));
```

Reply:
ok